invisible_cities/cities/components.py

```diff
@@ -198,6 +201,42 @@ def create_timestamp_(event_number: Union[int, float]) -> float:
     return create_timestamp_
 
 
+def run_git_command(git_command   : str,
+                    git_parameter : str = ""):
+    """
+    Runs a git command and stores the output.
+    
+    Parameters
+    ----------
+    git_command              : the standard terminal git command the user would input
+    git_parameter (Optional) : a string input to specify what git parameter the user is fetching
+
+    Returns
+    -------
+    The standard git terminal output
+    """
+    git_command = git_command.split() # turns the string input into a list of strings
+    try:
+        git_output = subprocess.run(git_command, text=True, capture_output=True, check=True).stdout.strip()
+        return git_output
+    except subprocess.CalledProcessError as e:
+        print(f"Following error encountered when running: " + ' '.join(git_command))
+        print(e.stderr)
+        return None
+    except Exception as e:
+        print(traceback.format_exc())
```

Comment:
Probably better to do `git_command.split()` inline in the call to `subprocess.run` so you don't have to join it back in the print below.

Reply:
right. Then you can return `""` or, even better, something like `"Unknown"` or `"None"`.